tests/test_minimizers.py

```diff
@@ -86,6 +122,121 @@ def test_powell(self):
+        a_vec = np.random.normal(15.0, scale=2.0, size=xdata.shape)
+        b_vec = np.random.normal(100, scale=2.0, size=xdata.shape)
+        ydata = a_vec * xdata + b_vec  # Point scattered around the line 5 * x + 105
+
+        # Normal symbolic fit
+        a = Parameter('a', value=0, min=0.0, max=1000)
+        b = Parameter('b', value=0, min=0.0, max=1000)
+
+        # Make a set of all ScipyMinimizers, and add a chained minimizer.
+        scipy_minimizers = subclasses(ScipyMinimize)
+        chained_minimizer = partial(ChainedMinimizer,
+                                    minimizers=[DifferentialEvolution, BFGS])
+        scipy_minimizers.add(chained_minimizer)
+        constrained_minimizers = subclasses(ScipyConstrainedMinimize)
+        # Test for all of them if they can be pickled.
+        for minimizer in scipy_minimizers:
+            if minimizer is MINPACK:
+                fit = minimizer(
+                    partial(chi_squared, x=xdata, y=ydata, sum=False),
+                    [a, b]
+                )
+            elif minimizer in constrained_minimizers:
+                # For constraint minimizers we also add a constraint, just to be
+                # sure constraints are treated well.
+                dummy_model = CallableNumericalModel({}, independent_vars=[], params=[a, b])
+                fit = minimizer(
+                    partial(chi_squared, x=xdata, y=ydata),
+                    [a, b],
+                    constraints=[Constraint(Ge(b, a), model=dummy_model)]
+                )
+            elif isinstance(minimizer, partial) and issubclass(minimizer.func, ChainedMinimizer):
+                init_minimizers = []
+                for sub_minimizer in minimizer.keywords['minimizers']:
+                    init_minimizers.append(sub_minimizer(
+                        partial(chi_squared, x=xdata, y=ydata),
+                        [a, b]
+                    ))
+                minimizer.keywords['minimizers'] = init_minimizers
+                fit = minimizer(partial(chi_squared, x=xdata, y=ydata), [a, b])
+            else:
+                fit = minimizer(partial(chi_squared, x=xdata, y=ydata), [a, b])
+
+            dump = pickle.dumps(fit)
+            pickled_fit = pickle.loads(dump)
+            problematic_attr = [
+                'objective', '_objective', 'wrapped_objective',
+                '_constraints', 'constraints', 'wrapped_constraints',
+                'local_minimizer', 'minimizers'
+            ]
+
+            for key, value in fit.__dict__.items():
+                new_value = pickled_fit.__dict__[key]
+                try:
+                    self.assertEqual(value, new_value)
+                except AssertionError as err:
+                    if key in problematic_attr:
+                        # These attr are new instances, and therefore do not
+                        # pass an equality test. All we can do is see if they
+                        # are at least the same type.
```

Comment:
I'm not sure, especially constraints are hard to compare because they use dummy variables for the dependent variables. But the proof is in the pudding: if the fits give the same result then the constraints were definitely imposed ;).

Reply:
Only if it would otherwise find a solution that violates the constraints. And since it's not binary reproducible (in general) I'm very hesitant to make tests chance dependant.

Maybe the number of constraints, and the number of variables/parameters per constraint?